docs/DeveloperGuide.md

```diff
@@ -1111,8 +1138,30 @@ and borrowed by a patron.
    and second book are borrowed by the first patron and requested by the second patron. Ensure that the third book is
    borrowed by the second patron.
 
-   3. Test case: `book related 2`<br>
-      Expected: The first, second and third book are shown in the book list. 
+   2. Test case: `book related 2`<br>
+      Expected: The first, second and third book are shown in the book list.
+
+### Searching for a book based on tags, author, title
+1. Searching for a book based on title.
+
+  1. Prerequisites: List all books using the `book list` command. Ensure that there is one book is titled "Harry Potter and the Philosopher's Stone".
+
+  2. Test case: `book find n/Harry Potter and the Philospher's Stone`
+     Expected: One book will be displayed in the book list. 
+
+2. Searching for a book based on tags.
+
+  1. Prerequisites: List all books using the `book list` command. Ensure that one book has the tag "Romance".
+
+  2. Test case: `book find t/Adventure`
+     Expected: One book will be shown in the book list.
```

Comment:
Try being a little more specific. "One book with tag romance will be shown in the book list"